### PR TITLE
[BUGFIX] merchandise spec addToCart tracking

### PR DESCRIPTION
**Pull request by torai55**
- tracking merchandise spec addToCart event
- use find instead of filter to avoid ts noUncheckedIndexedAccess bug

### CODE REVIEW

File: src/components/merchandise/MerchandisePaymentButton.tsx [hunk 1]
```diff
@@ -122,7 +131,14 @@ const GeneralMerchandisePaymentBlock: React.VFC<{
           colorScheme="primary"
           isFullWidth
           isDisabled={merchandise.isLimited && (quantity === 0 || quantity > remainQuantity)}
-          onClick={() => quantity && handleClick().then(() => history.push('/cart'))}
+          onClick={() => {
+            const resource = resourceCollection.filter(notEmpty)[0]
```
Comment (futianshen) [inline src/components/merchandise/MerchandisePaymentButton.tsx:135]: why not use [`find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) here ?

File: src/components/merchandise/MerchandisePaymentButton.tsx [hunk 2]
```diff
@@ -122,7 +131,14 @@ const GeneralMerchandisePaymentBlock: React.VFC<{
...
+            quantity &&
+              !isProductInCart?.('MerchandiseSpec', merchandiseSpec.id) &&
+              resource &&
+              tracking.addToCart(resource, { direct: true, quantity })
+            quantity && handleClick().then(() => history.push('/cart'))
```
Comment (futianshen) [inline src/components/merchandise/MerchandisePaymentButton.tsx:140]: NITS: I think code below is more clear and readable
```ts
if (quantity) {
  if( !isProductInCart?.('MerchandiseSpec', merchandiseSpec.id) && resource) {
    // do sth
  }
  // do sth
}
```

File: src/components/merchandise/MerchandisePaymentButton.tsx [hunk 3]
```diff
@@ -156,7 +176,11 @@ const CustomizedMerchandisePaymentBlock: React.VFC<{
           colorScheme="primary"
           isFullWidth
           isDisabled={(isAuthenticated && isLoading) || isPaymentButtonDisable}
-          onClick={() => (isAuthenticated ? onOpen?.() : setAuthModalVisible?.(true))}
+          onClick={() => {
+            isAuthenticated ? onOpen?.() : setAuthModalVisible?.(true)
+            const resource = resourceCollection.filter(notEmpty)[0]
+            resource && tracking.addToCart(resource, { direct: true, quantity })
```
Comment (futianshen) [inline src/components/merchandise/MerchandisePaymentButton.tsx:182]: same as above

File: src/components/merchandise/MerchandisePaymentButton.tsx (hunk 1 above)
Comment (torai55) [reply src/components/merchandise/MerchandisePaymentButton.tsx]: using `filter` will run into ts bug, noUncheckedIndexedAccess.
so `find` is better.

**Pull request merged**